plugin/trino-hive/src/main/java/io/trino/plugin/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -419,6 +419,40 @@ private static TableStatistics getTableStatistics(
         return result.build();
     }
 
+    private static TableStatistics createEmptyTableStatisticsWithPartitionColumnStatistics(
+            Map<String, ColumnHandle> columns,
+            Map<String, Type> columnTypes,
+            List<HivePartition> partitions)
+    {
+        TableStatistics.Builder result = TableStatistics.builder();
+        for (Map.Entry<String, ColumnHandle> column : columns.entrySet()) {
+            String columnName = column.getKey();
+            HiveColumnHandle columnHandle = (HiveColumnHandle) column.getValue();
+            Type columnType = columnTypes.get(columnName);
+            if (columnHandle.isPartitionKey()) {
+                result.setColumnStatistics(columnHandle, createPartitionColumnStatistics(columnHandle, columnType, partitions));
+            }
+        }
+        return result.build();
+    }
+
+    private static ColumnStatistics createPartitionColumnStatistics(HiveColumnHandle column, Type type, List<HivePartition> partitions)
+    {
+        if (partitions.isEmpty()) {
+            return ColumnStatistics.empty();
+        }
+
+        double estimatedNullsCount = partitions.stream()
```

Comment:
nit: Why it is double, not integer (what is expected for `count`)? I would cast it where we actually need double.

Reply:
We have to convert back to double below when setting the estimate